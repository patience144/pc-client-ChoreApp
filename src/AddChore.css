.chore-details {
  height: 25%;
  display: flex;
  flex-direction: column;
  margin-top: 14%;
}

main form {
  display: flex;
  justify-content: center;
  height: 100%;
  width: 50%;
  margin-left: 25%;
}

.chore-details input {
  line-height: 2.5em;
  border-radius: 3px;
  border: 1px solid rgba(0, 0, 0, 0.507);
  padding-left: 10px;
}

.chore-details button {
  font-size: 18px;
  padding: 8px 40px;
  margin-top: 20px;
}
