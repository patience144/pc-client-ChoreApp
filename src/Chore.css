.chore {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.chore a {
  width: 25px;
  text-align: center;
  text-decoration: none;
  border: 1px solid rgba(0, 0, 0, 0.445);
  padding: 5px 20px;
  background: rgb(189, 188, 189);
}

.chore input[type=checkbox] {
  align-self: center;
}

.chore label {
  width: 150px;
  word-wrap: break-word;
  padding-right: 5px;
}

.checked {
  text-decoration: line-through;
  color: gray;
}

.chore button {
  align-self: center;
  height: 28px;
}

.h3item {
  text-align: center;
}