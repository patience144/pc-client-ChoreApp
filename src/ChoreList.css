.chore-list {
  display: flex;
  justify-content: flex-end;
}