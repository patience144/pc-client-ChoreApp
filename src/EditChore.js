import React from 'react';
import { withRouter } from 'react-router-dom';
import ChoresContext from './context';
import Error from './Error';
import api from './api';
import './EditChore.css';

class EditChore extends React.Component {
    state = { chore: '' }

    static defaultProps = { chores: [] }

    static contextType = ChoresContext;

    onSubmitForm = (e) => {
        e.preventDefault();
        let chores = this.context.chores;
        const id = Number(this.props.match.params.choreID);
        const values = { name: e.target.name.value }
        chores = chores.map(chore => 
          (chore.chore_id === id)
            ? Object.assign(chore, values)
            : chore
        )
        this.context.updateChores(chores);
        this.props.history.push('/chores');
        api.editChore(id, values)
          .catch(error => <Error message={error} />)
    }

    render() {
      // const id = Number(this.props.match.params.choreID);
      // const foundChore = this.props.chores.find(chore => chore.chore_id === id);
        return (
          <form className='chore-details' onSubmit={this.onSubmitForm}> 
              <input
                  id='name'
                  type='text'
                  name='chore' 
                  onChange={e => this.setState({chore: e.target.value})}
                  value={this.state.chore}
              />
              
              <button type='submit'>Save</button>
              
          </form>
        );
    };

    componentDidMount() {
      const id = Number(this.props.match.params.choreID);
      const chore = this.props.chores.find(chore => chore.chore_id === id);
      const name = (chore) ? chore.name : null;
      this.setState({ chore: name })
    }
};

export default withRouter(EditChore);