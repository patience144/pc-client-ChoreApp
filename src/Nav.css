.nav {
  display: flex;
  justify-content: space-between;
  padding: 0 20px;
  border: 1px solid rgba(0, 0, 0, 0.26);
}

.nav a {
  text-decoration: none;
}

/* .nav-list a {

} */