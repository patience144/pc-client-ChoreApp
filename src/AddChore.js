import React from 'react';
import { withRouter } from 'react-router-dom';
import ChoresContext from './context';
import Error from './Error';
import api from './api';
import './AddChore.css';

class AddChore extends React.Component {
    state = { chore: '' }

    static defaultProps = { chores: [] }

    static contextType = ChoresContext;

    onSubmitForm = (e) => {
        e.preventDefault();
        const chores = this.context.chores
        const chore = { 
            name: e.target.name.value
        };
        api.createChore(chore)
          .then(chore => {
            chores.push(chore);
            this.context.updateChores(chores);
            this.props.history.push('/chores');
          })
          .catch(error => <Error message={error} />)

    }

    render() {
      // const id = Number(this.props.match.params.choreID);
      // const foundChore = this.props.chores.find(chore => chore.chore_id === id);
        return (
          <form className='chore-details' onSubmit={this.onSubmitForm}> 
              <input
                  id='name'
                  type='text'
                  name='chore' 
                  onChange={e => this.setState({chore: e.target.value})}
                  placeholder={'Add Chore'}
                  value={this.state.chore}
              />
              
              <button type='submit'>Add</button>
              
          </form>
        );
    };

    componentDidMount() {
      const id = Number(this.props.match.params.choreID);
      const chore = this.props.chores.find(chore => chore.chore_id === id);
      const name = (chore) ? chore.name : '';
      this.setState({ chore: name })
    }
};

export default withRouter(AddChore);