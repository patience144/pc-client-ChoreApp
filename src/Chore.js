import React from "react";
import { withRouter } from "react-router-dom";
import ChoresContext from "./context";
import Error from "./Error";
import api from "./api";
import "./Chore.css";

class Chore extends React.Component {
  state = { completed: this.props.chore.completed };

  static defaultProps = { chore: {} };

  static contextType = ChoresContext;

  deleteChore = (id) => {
    const chores = this.context.chores.filter((chore) => chore.chore_id !== id);
    this.context.updateChores(chores);
    api.deleteChore(id);
  };

  toggleComplete(checked) {
    const id = Number(this.props.chore.chore_id);
    const values = { completed: checked };
    const chores = this.context.chores.map((chore) => {
      if (chore.chore_id === id) Object.assign(chore, values);
      return chore;
    });
    this.context.updateChores(chores);
    api.editChore(id, values).catch((error) => <Error message={error} />);
  }

  render = () => {
    return (
      <div className='chore'>
        <input
          onFocus={(evt) => {
            evt.target.style.color = "red";
          }}
          onChange={(e) => {
            this.setState({ completed: e.target.checked });
            this.toggleComplete(e.target.checked);
          }}
          type='checkbox'
          name='completed'
          checked={this.props.chore.completed}
        />
        <label className={this.props.chore.completed ? "checked" : null}>
          {this.props.chore.name}
        </label>
        <button
          onClick={() =>
            this.props.history.push(`/edit/${this.props.chore.chore_id}`)
          }
          className='item-button'
        >
          Edit
        </button>
        <button
          className='item-button'
          onClick={() => this.deleteChore(this.props.chore.chore_id)}
        >
          Delete
        </button>
      </div>
    );
  };
}

export default withRouter(Chore);
