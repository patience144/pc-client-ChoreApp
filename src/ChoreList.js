import React from 'react';
import Chore from './Chore';
import './ChoreList.css';

class ChoreList extends React.Component {
    static defaultProps = { chores: [] }

    render() {
        return (
            <div className="chore-list">
                <h3 className='h3item'>Chores</h3>
                    <table>
                      <tbody>
                        {
                            this.props.chores.map(chore => {
                                return (
                                    <tr>
                                        <td>
                                            <Chore key={chore.chore_id} chore={chore} />
                                        </td>
                                    </tr>
                                );
                            })
                        }
                      </tbody>
                    </table>
            </div>
        )
    }
}

export default ChoreList;