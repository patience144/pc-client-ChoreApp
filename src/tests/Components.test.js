import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import AddChore from '../AddChore';
import Chore from '../Chore';
import ChoreList from '../AddChore';
import EditChore from '../EditChore';
import Nav from '../Nav';
import Welcome from '../Welcome';

describe('Main Section', () => {
  const addChore = <BrowserRouter><AddChore /></BrowserRouter>;
  const chore = <BrowserRouter><Chore /></BrowserRouter>;
  const choreList = <BrowserRouter><ChoreList /></BrowserRouter>;
  const editChore = <BrowserRouter><EditChore /></BrowserRouter>;
  const nav = <BrowserRouter><Nav /></BrowserRouter>;
  const welcome = <BrowserRouter><Welcome /></BrowserRouter>;
  describe('The Chore Details (Add) Page', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(addChore, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(addChore).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Chore component', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(chore, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(chore).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Chore List', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(choreList, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(choreList).toJSON();
      expect(tree).toMatchSnapshot();
    });
    describe('The Chore Details (Edit) Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(editChore, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(editChore).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('The Nav', () => {
    it('render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(nav, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(nav).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Welcome Page', () => {
    it('render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(welcome, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
      const tree = renderer.create(welcome).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});