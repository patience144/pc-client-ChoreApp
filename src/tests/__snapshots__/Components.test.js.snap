// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Section The Chore Details (Add) Page renders the UI as expected 1`] = `
<form
  className="chore-details"
  onSubmit={[Function]}
>
  <input
    id="name"
    name="chore"
    onChange={[Function]}
    placeholder="Add Chore"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;

exports[`Main Section The Chore List The Chore Details (Edit) Page renders the UI as expected 1`] = `
<form
  className="chore-details"
  onSubmit={[Function]}
>
  <input
    id="name"
    name="chore"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <button
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`Main Section The Chore List renders the UI as expected 1`] = `
<form
  className="chore-details"
  onSubmit={[Function]}
>
  <input
    id="name"
    name="chore"
    onChange={[Function]}
    placeholder="Add Chore"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;

exports[`Main Section The Chore component render the UI as expected 1`] = `
<div
  className="chore"
>
  <input
    name="completed"
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
  />
  <label
    className={null}
  />
  <button
    className="item-button"
    onClick={[Function]}
  >
    Edit
  </button>
  <button
    className="item-button"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;

exports[`Main Section The Nav render the UI as expected 1`] = `
<div
  className="nav"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Chore App
    </h1>
  </a>
  <ul
    className="nav-list"
  >
    <li>
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      >
        About
      </a>
    </li>
    <li>
      <a
        href="/chores"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      >
        Chore List
      </a>
    </li>
    <li>
      <a
        href="/add"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      >
        Add Chores
      </a>
    </li>
  </ul>
</div>
`;

exports[`Main Section The Welcome Page render the UI as expected 1`] = `
<div
  className="welcome"
>
  <h2>
    Let's make Chores Easier!!!
  </h2>
  <h3>
    Welcome to the Chore App A place where you can create a list of your chores and work throug them digitally! You can add a chore, delete a chore, or simply check it off for the day! Ready to begin?
  </h3>
  <a
    href="/add"
    onClick={[Function]}
  >
    Click Here to Start!
  </a>
</div>
`;
